CONTRIBUTING.md

```diff
@@ -0,0 +1,134 @@
+
+To run tests, run `./bin/phpunit -c ./Build/BuildEssentials/PhpUnit/UnitTests.xml` for unit tests or `./bin/phpunit -c ./Build/BuildEssentials/PhpUnit/FunctionalTests.xml` for functional/integration tests.
+
+---
+
+**Note**
+
+We use an upmerging strategy: create all bugfixes to the lowest maintained branch that contains the issue. Typically, this is the second last LTS release - see the diagram at https://www.neos.io/features/release-process.html.
+
+  For new features, pull requests should be made against the branch for the next minor version (named like `x.y`). Breaking changes must only go into the branch for the next major version.
+
+---
+
+For more detailed information, see https://discuss.neos.io/t/development-setup/504,
+https://discuss.neos.io/t/creating-a-pull-request/506 and
+https://discuss.neos.io/t/git-branch-handling-in-the-neos-project/6013
+
+
+## Neos 9
+
+
+### Prerequisites
+
+- You need PHP 8.2 installed.
+- Please be sure to run off the Neos development distribution in branch 9.0, to avoid dependency issues.
+
+### Setup
+
+The Event-Sourced Content Repository has a Docker Compose file included in `Neos.ContentRepository.BehavioralTests` which starts both Mariadb and Postgres in compatible versions. Additionally, there exists a helper to change the configuration in your distribution (`/Configuration`) to the correct values matching this database.
+
+Do the following for setting up everything:
+
+1. Start the databases:
+
+   ```shell
+   pushd Packages/Neos/Neos.ContentRepository.BehavioralTests; docker compose up -d; popd
+
+   # to stop the databases:
+   pushd Packages/Neos/Neos.ContentRepository.BehavioralTests; docker compose down; popd
+   # to stop the databases AND remove the stored data:
+   pushd Packages/Neos/Neos.ContentRepository.BehavioralTests; docker compose down -v; popd
+   ```
+
+2. Copy matching Configuration:
+
+   ```shell
+   cp -R Packages/Neos/Neos.ContentRepository.BehavioralTests/DistributionConfigurationTemplate/* Configuration/
+   ```
+
+3. Run Doctrine Migrations:
+
+   ```shell
+   ./flow doctrine:migrate
+   FLOW_CONTEXT=Testing/Postgres ./flow doctrine:migrate
+   ```
+
+4. Setup the Content Repository
+
+   ```shell
+   ./flow cr:setup
+   ```
+
+5. Set up Behat
+
+   ```shell
+   cp -R Packages/Neos/Neos.ContentRepository.BehavioralTests/DistributionBehatTemplate/ Build/Behat
+   pushd Build/Behat/
+   rm composer.lock
+   composer install
+   popd
+   ```
+
+### Running the Tests
+
+The normal mode is running PHP locally, but running Mariadb / Postgres in containers (so we know
+we use the right versions etc).
+
+```shell
+bin/behat -c Packages/Neos/Neos.ContentRepository.BehavioralTests/Tests/Behavior/behat.yml.dist
+```
+
+Running all tests can take a long time, depending on the hardware.
+To speed up the process, Behat tests can be executed in a "synchronous" mode by setting the `CATCHUPTRIGGER_ENABLE_SYNCHRONOUS_OPTION` environment variable:
```

Comment:
The `CATCHUPTRIGGER_ENABLE_SYNCHRONOUS_OPTION` parts in this readme are new and the flag is actually not required anymore as we are moving to sync by default see https://github.com/neos/neos-development-collection/issues/4746

Reply:
Doh. I _think_ I copied this from the existing README when I split it? 😳

Reply:
Updated based on the README content I replaced…